addons/binding/org.openhab.binding.lightify/src/main/java/org/openhab/binding/lightify/internal/link/LightifyLink.java

```diff
@@ -0,0 +1,426 @@
+            int zoneId = buffer.getShort();
+
+            byte[] nameBuffer = new byte[16];
+            buffer.get(nameBuffer, 0, nameBuffer.length);
+            String name = new String(nameBuffer, CP437).trim();
+
+            LightifyZone zone = new LightifyZone(this, name, zoneId);
+            zones.put(getZoneUID(zoneId), zone);
+            performZoneInfo(zone, consumer);
+        }
+    }
+
+    private void handleStatusUpdate(ByteBuffer buffer, Consumer<LightifyLuminary> consumer) {
+        // skip header
+        buffer.get();
+        if (STATUS_SINGLE.getId() != buffer.get()) {
+            throw new IllegalStateException("Illegal packet type");
+        }
+
+        buffer.getShort(); // sequence number
+        buffer.get(); // always 0
+        buffer.get(); // always 0
+        buffer.get(); // always 0
+
+        int id = buffer.getShort();
+
+        byte[] address = new byte[8];
+        buffer.get(address, 0, address.length);
+        String mac = DatatypeConverter.printHexBinary(address);
+        LightifyLuminary luminary = findDevice(mac);
+        if (luminary != null) {
+            buffer.getShort(); // unk1
+
+            boolean status = buffer.get() == 1;
+            byte luminance = buffer.get();
+            short temperature = buffer.getShort();
+            byte r = buffer.get();
+            byte g = buffer.get();
+            byte b = buffer.get();
+            byte a = buffer.get();
+
+            luminary.updatePowered(status);
+            luminary.updateLuminance(luminance);
+            luminary.updateTemperature(temperature);
+            luminary.updateRGB(r, g, b);
+
+            consumer.accept(luminary);
+        }
+    }
+
+    private void handleLightResponse(ByteBuffer buffer, Command command, Consumer consumer) {
+        // skip header
+        buffer.get();
+        if (command.getId() != buffer.get()) {
+            throw new IllegalStateException("Illegal packet type");
+        }
+
+        buffer.getShort(); // sequence number
+        buffer.get(); // always 0
+        buffer.get(); // always 0
+        buffer.get(); // always 0
+
+        int id = buffer.getShort();
+
+        byte[] address = new byte[8];
+        buffer.get(address, 0, address.length);
+        String mac = DatatypeConverter.printHexBinary(address);
+        LightifyLuminary luminary = findDevice(mac);
+        if (luminary != null) {
+            consumer.accept(luminary);
+        }
+    }
+
+    private void handleStatusAll(ByteBuffer buffer, Consumer consumer) {
+        // skip header
+        buffer.get();
+        if (STATUS_ALL.getId() != buffer.get()) {
+            throw new IllegalStateException("Illegal packet type");
+        }
+
+        buffer.getShort(); // sequence number
+        buffer.get(); // always 0
+        buffer.get(); // always 0
+        buffer.get(); // always 0
+
+        int numOfLights = buffer.getShort();
+        LOGGER.info("Found " + numOfLights + " devices...");
```

Comment:
1. Pelase change logging level to `debug`
2. Please use parametrized logging instead of string concatenation.